### PR TITLE
feat(router): Add defaultQueryParamsHandling to router configuration

### LINKED ISSUE #12664: Allow router configuration so that preserveQueryParams is true by default
Currently the router allows an individual navigation to preserve the query params (eg. Google's common `hl` query param)
This feature request is to allow the router to be configured so that this behavior takes place for all navigation.

### PR DESCRIPTION
**Pull request by atscott**
This commit adds an option to specify the default value for `queryParamsHandling` in `Router.createUrlTree` when another option is not specified (or is `null|undefined`).

resolves #12664

### CODE REVIEW

File: packages/router/src/router_config.ts [hunk 1]
```diff
@@ -8,7 +8,7 @@
 
 import {InjectionToken} from '@angular/core';
 
-import {OnSameUrlNavigation} from './models';
+import {OnSameUrlNavigation, QueryParamsHandling} from './models';
```
Comment (thePunderWoman) [inline packages/router/src/router_config.ts:11]: Usually there's also shipping to go along with the handling... maybe I'm mistaken.

**Review by thePunderWoman — APPROVED** [review]
reviewed-for: public-api

File: packages/router/src/router_config.ts (hunk 1 above)
Comment (atscott) [reply packages/router/src/router_config.ts:11]: This isn't included in our logistics solution right now. Maybe open a feature request?

File: packages/router/test/create_url_tree.spec.ts [hunk 2]
```diff
@@ -574,6 +575,43 @@ describe('createUrlTree', async () => {
   });
 });
 
+describe('defaultQueryParamsHandling', () => {
```
Comment (AndrewKushnir) [inline packages/router/test/create_url_tree.spec.ts:578]: I'd propose adding a test (to the necessary `describe` block) that would test the newly-added `"replace"` option.

**Review by AndrewKushnir — APPROVED** [review]
Reviewed-for: public-api, fw-router

**Comment by thePunderWoman** [comment]
This PR was merged into the repository by commit 6c76c91e151b53dfaccb4be43d43a8d857715dd7.

The changes were merged into the following branches: main

File: packages/router/test/create_url_tree.spec.ts [hunk 3]
```diff
@@ -574,6 +575,45 @@ describe('createUrlTree', async () => {
...
+    return router;
+  }
+
+  it('can use "merge" as the default', async () => {
+    const router = await setupRouter('merge');
+    await router.navigate(['new'], {queryParams: {'b': 2}});
+    expect(router.url).toEqual('/new?a=1&b=2');
+  });
+
+  it('can use "perserve" as the default', async () => {
```
Comment (davayd) [inline packages/router/test/create_url_tree.spec.ts:602]: "perserve" - "preserve" :)

**Pull request closed without merging**